src/main/scala/NIC.scala

```diff
@@ -273,27 +292,64 @@ class IceNicRecvPathModule(outer: IceNicRecvPath)
   val io = IO(new Bundle {
     val recv = Flipped(new IceNicRecvIO)
     val in = Flipped(Decoupled(new StreamChannel(NET_IF_WIDTH))) // input stream 
-    val tap = outer.tapFuncs.nonEmpty.option(
-      Vec(outer.tapFuncs.length, Decoupled(new StreamChannel(NET_IF_WIDTH))))
+    val tap = Vec(outer.tapFuncs.length, Decoupled(new StreamChannel(NET_IF_WIDTH)))
     val csum = checksumOffload.option(new Bundle {
       val res = Decoupled(new TCPChecksumOffloadResult)
       val enable = Input(Bool())
     })
-    val buf_free = Output(UInt(8.W))
+    val buf_free = Output(Vec(1 + outer.tapFuncs.length, UInt(8.W)))
```

Comment:
Nit: Might want to write a small comment to note that this is primary buffer + tap buffers